### DISCUSSION TITLE
interesting how do breakpoints properties directly?

### DISCUSSION THREAD

**Discussion by synasapmob** [chakra-ui/chakra-ui#8724] [General] [answered]
as the title, I would like to know **mechanism** how you do that. I meaning write ```<Box width={[1, 1 / 2, 1 / 4]} />``` 
and then you'll generate ```@media screen and (min-width: 30em)....```

I am curious about that, because maybe you were stuck on how to **SSR** in some cases also wish for the website to smooth 
so I want to learn about it and I'll do **SSR** as like **short length** of the string with **breakpoints**
doesn't matter warning to me **"Text content does not match server-rendered HTML"**

## Examples:
![Screen Shot 2024-08-09 at 10 38 33](https://github.com/user-attachments/assets/fa21dea3-83e2-4abf-918e-7ff79f8f463e)
![Screen Shot 2024-08-09 at 10 42 54](https://github.com/user-attachments/assets/156979af-93ae-42e1-93f5-aff43655115b)

## Correlation:
- [https://nextjs.org/docs/messages/react-hydration-error](https://nextjs.org/docs/messages/react-hydration-error)
- [https://v2.chakra-ui.com/docs/styled-system/responsive-styles#more-examples](https://v2.chakra-ui.com/docs/styled-system/responsive-styles#more-examples)

**Comment by segunadebayo** [comment, marked as answer]
The array is compiled to CSS `@media` queries. Nothing reads the viewport in JS, so SSR stays in sync.

`width={[1, 1/2, 1/4]}` becomes `base`, `sm`, `md` in order. Object syntax is the same thing: `{ base, sm, md }`.
